src/HorusStudio.Maui.MaterialDesignControls/Controls/MaterialLabel.cs

```diff
@@ -76,24 +128,26 @@ public new Color TextColor
 
         public MaterialLabel()
         {
-            base.FontFamily = this.FontFamily;
             base.TextColor = this.TextColor;
-            base.FontSize = MaterialFontSize.BodyMedium;
+
+            SetBinding(Label.FontFamilyProperty, new Binding(nameof(FontFamily), source: this));
+            SetBinding(Label.FontFamilyProperty, new Binding(nameof(FontFamilyRegular), source: this));
+            SetBinding(Label.FontFamilyProperty, new Binding(nameof(FontFamilyMedium), source: this));
+            SetBinding(Label.TextColorProperty, new Binding(nameof(TextColor), source: this));
+
+            if (Type == DefaultType)
+            {
+                TypeChanged(Type);
+            }
         }
 
         #endregion Constructors
 
         #region Methods
 
-        private static void OnTypeChanged(BindableObject bindable, object oldValue, object newValue)
-        {
-            var control = (MaterialLabel)bindable;
-            control.ApplyTypeProperty((LabelTypes)newValue);
-        }
-
-        private void ApplyTypeProperty(LabelTypes labelType)
+        private void TypeChanged(LabelTypes type)
         {
-            switch (labelType)
+            switch (type)
```

Comment:
I know it's just a refactor but maybe we should leave a TODO comment to check this characterspacings for each type according to Material